### ISSUE TITLE
Files with åäö in there name gets an inputstream that is null in MultipartForm uploads

### ISSUE THREAD

**Issue by bjornpalmqvist** [quarkusio/quarkus#41272]
### Describe the bug

When reciving multipart form data that conatins a file represented as an inputstream. The input stream is null if the filename contains åäö (in utf-8). This worked in the previous version of quarkus.

### Expected behavior

Files with eg. åäö is not null.

### Actual behavior

Files with eg. åäö is null.

### How to Reproduce?

I attached a smal tgz that repruduces this case.
[multipart.zip](https://github.com/user-attachments/files/15881984/multipart.zip)

### Output of `uname -a` or `ver`

Linux 7540-0119 5.15.153.1-microsoft-standard-WSL2 #1 SMP Fri Mar 29 23:14:13 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

openjdk 21.0.2 2024-01-16 LTS

### Quarkus version or git rev

3.8.4

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.6 (bc0240f3c744dd6b6ec2920b3cd08dcc295161ae)

**Comment by geoand** [comment]
Is this still an issue?

Also, do you know which was the last version of Quarkus that this worked?

**Comment by gsmet** [comment]
From what I can see, it's still an issue with 3.18.3.

@jamezp does it ring a bell? The reproducer is using RESTEasy.

**Comment by jamezp** [comment]
@gsmet Yes. There was a bug in Apache James mim4j which was recently fixed and released. The RESTEasy issue was https://issues.redhat.com/browse/RESTEASY-3495 and it was fixed in mime4j 0.8.12.

**Comment by gsmet** [comment]
@jamezp oh, thanks a lot, I can confirm bumping Mime4j in https://github.com/quarkusio/quarkus/pull/46344 fixes the issue.
